src/odemis/driver/xt_client.py

```diff
@@ -366,3 +365,362 @@ def set_beam_shift(self, x_shift, y_shift):
     def beam_shift_info(self):
         """Returns: (dict) the unit and xy-range of the beam shift."""
         return self.client.beam_shift_info()
+
+
+class SEM(model.HwComponent):
+    """
+    Connects to a Zeiss SEM via the RemCon interface (over RS-232).
+    At initialisation, the SEM software should already be running, and the
+    RemCon option active (might require an extra license).
+    """
+
+    def __init__(self, name, role, children, address, timeout, daemon=None,
+                 **kwargs):
+        """
+        Parameters
+        ----------
+        address: str
+            server address and port of the Microscope server.
+        """
+
+        model.HwComponent.__init__(self, name, role, daemon=daemon, **kwargs)
+        self._swVersion = "xtlib"  # TODO figure out sw version
+        self._hwVersion = "Unknown"  # TODO add get_InstrumentID to xt adapter
+        # connect to the microscope pc using zerorpc
+        self.client = zerorpc.Client(address, heartbeat=None, timeout=timeout)
```

Comment:
I didn't think of it in this way. I thought you'd create a "MicroscopeClient", and use it afterwards. That said, as almost every function is a direct mapping, you're right that it's probably as easy to directly call the remote methods. That said, I'd suggest then to (eventually) delete the `MicroscopeClient` class, to avoid code duplication.
Note that a third alternative would be to move all the methods of `MicroscopeClient` onto `SEM` (ie, essentially merging the two classes). Then the scanner/stage could do `self.parent.method()`. The main advantage of this, is that if you want to debug, it's actually possible to call the method on the `SEM` from a terminal, which is the not the case if you have a sub-component.

Reply:
I was in doubt between what I implemented and creating a `MicroscopeClient` and use that afterwards. But now I actually like the last option, so I'll merge the `SEM` and the `MicroscopeClient` classes